doc/api/zlib.markdown

```diff
@@ -241,6 +242,10 @@ ignored by the decompression classes.
 See the description of `deflateInit2` and `inflateInit2` at
 <http://zlib.net/manual.html#Advanced> for more information on these.
 
+By default, the zlib methods will throw an error when decompressing truncated
+data. If you know what you are doing, you can override this behaviour by passing
```

Comment:
The `If you know what you are doing...` part is a bit concerning here. Rather than this, perhaps include an example of using Z_SYNC_FLUSH and provide some detail about why someone would do so.